src/ops/fuelup_check.rs

```diff
@@ -160,3 +86,86 @@ pub fn check(command: CheckCommand) -> Result<()> {
     }
     Ok(())
 }
+
+fn check_toolchain(toolchain: &str, verbose: bool) -> Result<()> {
+    let latest_versions = match Channel::from_dist_channel(&DistToolchainName::from_str(toolchain)?)
+    {
+        Ok(c) => collect_versions(c),
+        Err(e) => {
+            error!(
+                "Failed to get latest channel {} - fetching versions using GitHub API",
+                e
+            );
+            [component::FORC, component::FUEL_CORE, component::FUELUP]
+                .iter()
+                .map(|&c| {
+                    (
+                        c.to_owned(),
+                        DownloadCfg::new(c, target_from_name(c).ok(), None)
+                            .unwrap()
+                            .version,
+                    )
+                })
+                .collect()
+        }
+    };
+
+    let toolchain = Toolchain::from(toolchain)?;
+    bold(|s| writeln!(s, "{}", &toolchain.name));
+    for component in [component::FORC, component::FUEL_CORE] {
+        let component_executable = toolchain.path.join(component);
+
+        match std::process::Command::new(&component_executable)
+            .arg("--version")
+            .output()
+        {
+            Ok(o) => {
+                let version = Version::parse(
+                    String::from_utf8_lossy(&o.stdout)
+                        .split_whitespace()
+                        .nth(1)
+                        .expect("expected version"),
```

Comment:
I realise we used `expect` previously too, but perhaps it's worth using `ok_or_else` with `anyhow!` instead?

Reply:
Updated the error handling for this to print an error message instead of aborting.